static/js/settings_users.js

```diff
@@ -116,36 +108,71 @@ function populate_users(realm_people_data) {
+            return templates.render("admin_user_list", { user: item, can_modify: page_params.is_admin });
+        },
+        filter: {
+            element: $bots_table.closest(".settings-section").find(".search"),
+            callback: function (item, value) {
+                return (
+                    item.full_name.toLowerCase().match(value) ||
+                    item.email.toLowerCase().match(value)
+                );
+            },
+        },
+    }).init();
+
+    var $users_table = $("#admin_users_table");
+    list_render($users_table, active_users, {
+        name: "users_table_list",
+        modifier: function (item) {
+            var activity_rendered;
+            if (people.is_current_user(item.email)) {
+                activity_rendered = timerender.render_date(new XDate());
+            } else if (presence.presence_info[item.user_id]) {
+                // XDate takes number of milliseconds since UTC epoch.
+                var last_active = presence.presence_info[item.user_id].last_active * 1000;
+                activity_rendered = timerender.render_date(new XDate(last_active));
             } else {
                 activity_rendered = $("<span></span>").text(i18n.t("Unknown"));
             }
-        }
-        row.find(".last_active").append(activity_rendered);
-        users_table.append(row);
-    });
 
-    var deactivated_table_html = "";
-    _.each(deactivated_users, function (user) {
-        var user_html = templates.render("admin_user_list", {user: user, can_modify: page_params.is_admin});
-        deactivated_table_html = deactivated_table_html.concat(user_html);
-    });
-    deactivated_users_table.append(deactivated_table_html);
+            var $row = $(templates.render("admin_user_list", {user: item, can_modify: page_params.is_admin}));
+
+            $row.find(".last_active").append(activity_rendered);
+
+            return $row;
+        },
+        filter: {
+            element: $users_table.closest(".settings-section").find(".search"),
+            callback: function (item, value) {
+                return (
+                    item.full_name.toLowerCase().match(value) ||
+                    item.email.toLowerCase().match(value)
+                );
+            },
+        },
+    }).init();
+
+    var $deactivated_users_table = $("#admin_deactivated_users_table");
+    list_render($deactivated_users_table, deactivated_users, {
+        name: "deactivated_users_table_list",
+        modifier: function (item) {
+            return templates.render("admin_user_list", { user: item });
+        },
+        filter: {
+            element: $deactivated_users_table.closest(".settings-section").find(".search"),
+            callback: function (item, value) {
+                return (
+                    item.full_name.toLowerCase().match(value) ||
+                    item.email.toLowerCase().match(value)
+                );
+            },
+        },
+    }).init();
```

Comment:
The issue appears to predate your changes, but this heading area is broken:
![image](https://cloud.githubusercontent.com/assets/2746074/25301590/3fe344c6-26e0-11e7-9b94-331605f858ad.png)